internal/controlplane/handlers_githubwebhooks.go

```diff
@@ -806,10 +806,16 @@ func (s *Server) processPackageEvent(
 		return nil, fmt.Errorf("error extracting artifact from payload: %w", err)
 	}
 
-	pbArtifact, err := ghprop.ArtifactV1FromProperties(refreshedPkgProperties)
+	ewp := models.NewEntityWithProperties(*ei, refreshedPkgProperties)
+	pbMsg, err := s.props.EntityWithPropertiesAsProto(ctx, ewp, s.providerManager)
 	if err != nil {
 		return nil, fmt.Errorf("error converting artifact to protobuf: %w", err)
 	}
+
+	pbArtifact, ok := pbMsg.(*pb.Artifact)
```

Comment:
this typing is a bit awkward, but as a follow-up (that I already have in another branch), I wanted to add a more generic WithEntityID that would replace WithRepositoryID and WithEntity that would replace WithRepository etc.